gips/0083.md

```diff
@@ -0,0 +1,250 @@
+
+Substreams endpoints provide signed attestations with each block's response, as per [the `attestation` field](https://github.com/streamingfast/substreams/blob/develop/proto/sf/substreams/rpc/v2/service.proto#L96) in the `BlockScopedData` response, matching the operator key specified in [the `attestation_public_key` field](https://github.com/streamingfast/substreams/blob/develop/proto/sf/substreams/rpc/v2/service.proto#L106) of the `SessionInit` response.
+
+The attestation payload MUST be signed by a special _attestation key_, derived from the operator's key, so that disputes can be opened on a separate allocation destined for disputes.
+
+A valid _attestation key_, attached to an allocation MUST be verified by the Payment Gateway, before giving the assurance to Consumers that the Indexer can be slashed for misbehaving.
+
+An Arbiter MUST be able to validate payloads, through  multiple providers and investigation, and after judgement, be able to slash the Indexer's stake, using the information provided in those payloads via an on-chain allocations.
+
+
+#### Methods of analysis
+
+THIS SECTION IS NON-NORMATIVE
+
+This section specifies potential methods an Arbiter can use to analyze disputes. Other more optimal tools could be written, and are out of scope of this specification.
+
+**Detecting Faults:** Discrepancies in data returned to a Fisherman or a Consumer, by different providers for the same stream and block, is material to open an investigation.
+
+The Arbiter can use the `substreams gui` tool or the `substreams run --show-attestations` tool to inspect attestations and contents, and compare stream data at specific block ranges, verifying data integrity and determinism.
+
+`substreams tools` will contain a way to validate those attestations, and help with investigation.  This tool will take the content being disputed (BlockScopedData) from the replay.log, which contain the attestations over the content within, and validate their output, in order to compare against providers, and end up slashing an Indexer.
+
+```
+substreams tools validate-attestation --indexer-address 0x123123123 --attestation eth:120398102398102938109283012983 --replay-log-file ./path/to/replay.log
+```
+
+This command will process all messages in the provided replay log and validate their attestations. Replay logs can be generated using  `substreams gui --with-replay`.
+
+The Payment Gateway can keep an in-memory mapping will link the operator key to the staking key.
+
+- The `setOperator` function ([Staking.sol#L226](https://github.com/graphprotocol/contracts/blob/ce3ec16484dacc89a1b9cf08455256be830790e3/packages/contracts/contracts/staking/Staking.sol#L226)) links the operator key to the staking key and emits the `SetOperator` event ([IStakingBase.sol#L99](https://github.com/graphprotocol/contracts/blob/main/packages/contracts/contracts/staking/IStakingBase.sol#L99)).
```

Comment:
This is also queryable in the Network Subgraph

Reply:
noted